enioka_scan/src/main/java/com/enioka/scanner/sdk/zebraoss/ssi/SsiMonoPacket.java

```diff
@@ -0,0 +1,157 @@
+package com.enioka.scanner.sdk.zebraoss.ssi;
+
+/**
+ * SSI packet, containing the header and checksum bytes. Suitable for both incoming and outgoing packets.
+ * Multi-packets are (so far) read-only and in a separate class as they require extra header elements.
+ */
+public final class SsiMonoPacket {
+    private static final String LOG_TAG = "SsiMonoPacket";
+    private final boolean ble;
+
+    // Packet attributes are in the same order as which they occur in the raw packet buffer
+    /** MSB of total length in little-endian used by BLE only, for outgoing packets only (these bytes should be skipped for reading). */
+    private byte blePacketLengthMsb;
+    /** LSB of total length in little-endian used by BLE only, for outgoing packets only (these bytes should be skipped for reading). */
+    private byte blePacketLengthLsb;
+    /** Length of the packet excluding the checksum. */
+    private byte packetLengthWithoutChecksum;
+    /** OpCode of the packet, describing the type of data it contains. */
+    private final byte opCode;
+    /** Source of the packet, to check against SsiSource helper. */
+    private final byte source;
+    /** Status of the packet, to check against SsiStatus helper. */
+    private final byte status;
+    /** Raw data encapsulated by the packet (excluding headers and checksum). */
+    private final byte[] data;
+    /** Checksum MSB. */
+    private byte checksumMsb;
+    /** Checksum LSB. */
+    private byte checksumLsb;
+
+    /**
+     * Creates a new SSI packet for receiving data.
+     * Only suitable for incoming packets. Checksum and length bytes will be verified and trigger an exception if invalid.
+     */
+    public SsiMonoPacket(final byte packetLengthWithoutChecksum,
+                         final byte opCode,
+                         final byte status,
+                         final byte[] data,
+                         final byte checksumMsb,
+                         final byte checksumLsb) {
+        if (opCode == SsiCommand.MULTIPACKET_SEGMENT.getOpCode())
+            throw new IllegalArgumentException("Opcode indicates a multipacket, not a monopacket.");
+
+        this.ble = false;
+
+        this.packetLengthWithoutChecksum = packetLengthWithoutChecksum;
+        this.opCode = opCode;
+        this.source = 0x00;
+        this.status = status;
+        this.data = data;
+        this.checksumMsb = checksumMsb;
+        this.checksumLsb = checksumLsb;
+
+        //this.validateLengthAndChecksum(); // FIXME - 2022/03/28: inbound packets are constantly rejected due to checksum. For now, assume they are all correct.
+    }
+
+    /**
+     * Creates a new SSI packet for sending commands.
+     * Only suitable for outgoing packets. Checksum and length are calculated automatically.
+     * @param ble Whether the packet must contain BLE length bytes or not.
+     */
+    public SsiMonoPacket(final byte opCode, final byte status, final byte[] data, final boolean ble) {
```

Comment:
I have a big issue with this `ble` new argument, that is only here to knwo if packets should be cut into smaller pieces with a header. This means that every command must know this, and therefore that any user of the command must know if BLE or not. This is not a concern for the caller really. We should find a way to pass that infomration without modifying the API of every command.